sig-installation/guideline/KubeSphere-terraform-zh.md

```diff
@@ -0,0 +1,180 @@
+sudo yum -y install terraform
+```
+* 是否安装成功及版本输出
+
+`terraform version`
+
+### 2、var.tf文件说明
+* access_key 和 secret_key 为必填项。点击 iaas 用户--》 API 密钥--》创建即可，然后把两个参数填入到下面的配置文件中。zone 可以根据自己需求来修改，默认 pek3a。
+
+```
+terraform {
+  required_providers {
+    qingcloud = {
+      source = "shaowenchen/qingcloud"
+      version = "1.2.6"
+    }
+  }
+}
+
+variable "access_key" {
+  default = "***"
+}
+
+variable "secret_key" {
+  default = "***"
+}
+
+variable "zone" {
+  default = "pek3a"
+}
+
+provider "qingcloud" {
+  access_key = "${var.access_key}"
+  secret_key = "${var.secret_key}"
+  zone = "${var.zone}"
+}
+```
+
+### 3、kubesphere.tf文件说明
+* resource qingcloud_eip 为创建外网ip，也可以用已存在外网 IP，如果用已存在外网 ip，就不需要 qingcloud_eip resource 模块。
+* qingcloud_security_group 为创建防火墙，也可以使用已存在防火墙，同理。
+* qingcloud_security_group_rule 为创建防火墙开放的端口。
+* qingcloud_keypair 为密钥创建，此处注释掉，用密码形式。
+* qingcloud_instance 创建机器，包含名字，操作系统，内存，cpu，磁盘，密码，绑定外网 IP，防火墙，子网和类型等。
+* null_resource 和 install_kubesphere 包含文件拷贝及执行命令。
+
+```
+resource "qingcloud_eip" "init"{
+  name = "tf_eip"
+  description = ""
+  billing_mode = "traffic"
+  bandwidth = 20
+  need_icp = 0
+}
+
+resource "qingcloud_security_group" "basic"{
+  name = "防火墙"
+  description = "这是第一个防火墙"
+}
+
+resource "qingcloud_security_group_rule" "openport" {
+  security_group_id = "${qingcloud_security_group.basic.id}"
+  protocol = "tcp"
+  priority = 0
+  action = "accept"
+  direction = 0
+  from_port = 22
+  to_port = 40000
+}
+
+# qingcloud_keypair upload an SSH public key
+# In this example, upload ~/.ssh/id_rsa.pub content.
+# You may not have this file in your system, you will need to create your own SSH key.
+#resource "qingcloud_keypair" "arthur"{
+#  name = "arthur"
+#  public_key = "${file("~/.ssh/id_rsa.pub")}"
+#}
+
+resource "qingcloud_instance" "init"{
+  count = 1
+  name = "master-${count.index}"
+  image_id = "centos76x64a"
+  cpu = "16"
+  memory = "32768"
+  instance_class = "0"
+  managed_vxnet_id="vxnet-0"
+#  keypair_ids = ["${qingcloud_keypair.arthur.id}"]
+  login_passwd = "Qcloud@123"
+  security_group_id ="${qingcloud_security_group.basic.id}"
+  eip_id = "${qingcloud_eip.init.id}"
+}
+
+resource "null_resource" "install_kubesphere" {
+  provisioner "file" {
+    destination = "./install.sh"
+    source      = "./install.sh"
+
+    connection {
+      type        = "ssh"
+      user        = "root"
+      host        = "${qingcloud_eip.init.addr}"
+      password    = "Qcloud@123"
+#      private_key = "${file("~/.ssh/id_rsa")}"
+      port        = "22"
+    }
+  }
+
+  provisioner "remote-exec" {
+    inline = [
+      "sh install.sh"
+    ]
+
+    connection {
+      type        = "ssh"
+      user        = "root"
+      host        = "${qingcloud_eip.init.addr}"
+      password    = "Qcloud@123"
+ #     private_key = "${file("~/.ssh/id_rsa")}"
+      port        = "22"
+    }
+  }
+}
+```
+
+### 4、install.sh 文件说明
+* 具体执行命令
+
+```
+curl -O -k https://kubernetes.pek3b.qingstor.com/tools/kubekey/kk
```

Comment:
This should be specified as the link to the release.

Reply:
Fixd